src/AST/Translator.Expressions.dfy

```diff
@@ -336,44 +340,44 @@ module Bootstrap.AST.Translator.Expressions {
 
   function method TranslateSeqSelectExpr(se: C.SeqSelectExpr): (e: TranslationResult<DE.T>)
     reads *
-    decreases ASTHeight(se), 0
+    decreases ASTHeight(se), 1
     ensures e.Success? ==> P.All_Expr(e.value, DE.WellFormed)
   { // FIXME: The models that we generate do not allow for `null`
     var ty :- TranslateType(se.Seq.Type);
-    :- Need(ty.Collection? && !ty.kind.Set?,
-            Invalid("`SeqSelect` must be on a map, sequence, or multiset."));
-    :- Need(se.SelectOne ==> se.E0 != null && se.E1 == null,
-            Invalid("Inconsistent values for `SelectOne` and E1 in SeqSelect."));
-    :- Need(!se.SelectOne ==> ty.kind.Seq?,
-            Invalid("`SeqSelect` on a map or multiset must have a single index."));
```

Comment:
Are these really ever violated?  I thought they were really invariants.

Reply:
Oh, yes, you're right.